SilKit/source/core/participant/Participant_impl.hpp

```diff
@@ -712,12 +741,16 @@ auto Participant<SilKitConnectionT>::CreateRpcClient(
 
     if (GetLogger()->GetLogLevel() <= Logging::Level::Trace)
     {
-        Logging::Trace(
-            GetLogger(),
-            "Created RPC Client '{}' with function name '{}' and media type '{}' for network '{}' with service name "
-            "'{}' and labels: {}",
-            controllerConfig.name, controllerConfig.functionName.value(), dataSpec.MediaType(), network,
-            controller->GetServiceDescriptor().to_string(), FormatLabelsForLogging(dataSpec.Labels()));
+        Logging::LoggerMessage lm{_logger.get(), Logging::Level::Trace};
+        lm.SetMessage("Created controller");
+        lm.SetKeyValue(Logging::Keys::controllerType, supplementalData[SilKit::Core::Discovery::controllerType]);
+        lm.SetKeyValue(Logging::Keys::controllerName, controllerConfig.name);
+        lm.SetKeyValue(Logging::Keys::controllerFuncName, controllerConfig.functionName.value());
+        lm.SetKeyValue(Logging::Keys::mediaType, dataSpec.MediaType());
+        lm.SetKeyValue(Logging::Keys::network, network);
+        lm.SetKeyValue(Logging::Keys::serviceName, controller->GetServiceDescriptor().to_string());
+        lm.SetKeyValue(Logging::Keys::label, FormatLabelsForLogging(dataSpec.Labels()));
```

Comment:
The `dataSpec.label` could be overwritten by the participant configuration, therefore use `configuredRpcSpec.Labels()` here. 
```suggestion
        lm.SetKeyValue(Logging::Keys::controllerFuncName, configuredRpcSpec.FunctionName());
        lm.SetKeyValue(Logging::Keys::mediaType, configuredRpcSpec.MediaType());
        lm.SetKeyValue(Logging::Keys::network, network);
        lm.SetKeyValue(Logging::Keys::serviceName, controller->GetServiceDescriptor().to_string());
        lm.SetKeyValue(Logging::Keys::label, FormatLabelsForLogging(configuredRpcSpec.Labels()));
```